src/gh/components/DF_preview_assembly/code.py

```diff
@@ -0,0 +1,52 @@
+#! python3
+
+import System
+
+import Rhino.Geometry as rg
+
+from ghpythonlib.componentbase import executingcomponent as component
+
+import diffCheck
+from diffCheck.df_geometries import DFAssembly
+import diffCheck.diffcheck_bindings
+import diffCheck.df_util
+
+import numpy as np
+
+
+class DFPreviewAssembly(component):
+    def __init__(self):
+        super(DFPreviewAssembly, self).__init__()
+        self._dfassembly = None
+
+    def RunScript(self, i_assembly: DFAssembly=None):
+        if i_assembly is None:
+            return None
+
+        self._dfassembly = i_assembly
+        return i_assembly.mass_center
+
+    # Preview overrides
+    def DrawViewportWires(self, args):
+        for beam in self._dfassembly.beams:
+            # beams' obb
+            df_cloud = diffCheck.diffcheck_bindings.dfb_geometry.DFPointCloud()
+            df_cloud.points = [np.array([vertex.Location.X, vertex.Location.Y, vertex.Location.Z]).reshape(3, 1) for vertex in beam.to_brep().Vertices]
+            obb: rg.Brep = diffCheck.df_cvt_bindings.cvt_dfOBB_2_rhbrep(df_cloud.get_tight_bounding_box())
+
+            # axis arrow
+            obb_faces = obb.Faces
+            obb_faces = sorted(obb_faces, key=lambda face: rg.AreaMassProperties.Compute(face).Area)
+            obb_endfaces = obb_faces[:2]
+            beam_axis = rg.Line(obb_endfaces[0].GetBoundingBox(True).Center, obb_endfaces[1].GetBoundingBox(True).Center)
+            extension_length = 0.5 * diffCheck.df_util.get_doc_2_meters_unitf()
+            beam_axis.Extend(extension_length, extension_length)
+            args.Display.DrawArrow(beam_axis, System.Drawing.Color.Magenta)
```

Comment:
I tried to append the lines in a list and use args.Display.DrawArrows(list_of_line) but it didn't help.

Reply:
As discussed today it seems to be a problem with not only the arrows but also lines but I truly think that having a better look at `deepcopy()` might help, we can also do this on Thursday together @DamienGilliard !

Reply:
solved it in [e9d0253](https://github.com/diffCheckOrg/diffCheck/pull/68/commits/e9d0253219811de7a350668c2448a87b428ad551): it was simply safer to use the native `copy`module and just do:
```py
o_assembly = copy.copy(i_assembly)
```
no more lag in the preview